sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/MicrosoftBotIdentifier.java

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.common;
+
+import com.azure.core.util.CoreUtils;
+
+/**
+ * Communication identifier for Microsoft Bot
+ */
+public final class MicrosoftBotIdentifier extends CommunicationIdentifier {
+    private final String botId;
+    private final boolean isGlobal;
+    private boolean rawIdSet = false;
+
+    private final CommunicationCloudEnvironment cloudEnvironment;
+
+    /**
+     * Creates a MicrosoftBotIdentifier object
+     *
+     * @param botId Id of the Microsoft bot.
+     * @param cloudEnvironment Cloud Environment of the Microsoft bot.
+     * @param isGlobal set this to true if the bot is global.
+     * @throws IllegalArgumentException thrown if botId parameter fail the validation.
+     */
+    public MicrosoftBotIdentifier(String botId, CommunicationCloudEnvironment cloudEnvironment, boolean isGlobal) {
+        if (CoreUtils.isNullOrEmpty(botId)) {
+            throw new IllegalArgumentException("The initialization parameter [botId] cannot be null or empty.");
+        }
+        this.botId = botId;
+        this.cloudEnvironment = cloudEnvironment;
+        this.isGlobal = isGlobal;
+        generateRawId();
+    }
+
+    /**
+     * Creates a MicrosoftBotIdentifier object
+     *
+     * @param botId Id of the Microsoft bot.
+     * @throws IllegalArgumentException thrown if botId parameter fail the validation.
+     */
+    public MicrosoftBotIdentifier(String botId) {
```

Comment:
what if developers wants to set a bot id & isGlobal, but not cloud env?

Reply:
If developer don't provide cloud env, by default it should be set to Public as we do for Microsoft Teams Identifier.
But yes, we should have another constructor accepting botId and isGlobal parameters

` public MicrosoftBotIdentifier(String botId, boolean isGlobal) 
  { `       
         `   this(botId, CommunicationCloudEnvironment.PUBLIC, isGlobal);`
  `}
`

Reply:
Added another constructor